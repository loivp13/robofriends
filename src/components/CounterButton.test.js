import { shallow, mount, render } from "enzyme";
import React from "react";
import CounterButton from "./CounterButton";

import { createSerializer } from "enzyme-to-json";
import toJson from "enzyme-to-json";
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

it("expect to render card list component", () => {
  const mockRobot = [
    { id: 1, name: "John Snow", username: "JohnJohn", email: "john@gmail.com" },
  ];
  expect(
    toJson(shallow(<CounterButton robots={mockRobot} />))
  ).toMatchSnapshot();
});

it("correctly increments the counter", () => {
  const mockColor = "red";
  const wrapper = shallow(<CounterButton color={mockColor} />);
  wrapper.find('[id="counter"]').simulate("click");
  expect(wrapper.state()).toEqual({ count: 1 });
  wrapper.find('[id="counter"]').simulate("click");
  expect(wrapper.state()).toEqual({ count: 2 });
  wrapper.find('[id="counter"]').simulate("click");
  expect(wrapper.state()).toEqual({ count: 3 });
  expect(wrapper.props().color).toEqual("red");
});
