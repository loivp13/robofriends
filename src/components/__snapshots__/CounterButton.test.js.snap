// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render card list component 1`] = `
<button
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
